pkg/core/native/native_gas.go

```diff
@@ -80,15 +81,20 @@ func (g *GAS) Initialize(ic *interop.Context) error {
 
 // OnPersist implements Contract interface.
 func (g *GAS) OnPersist(ic *interop.Context) error {
-	//for _ ,tx := range ic.block.Transactions {
-	//	g.burn(ic, tx.Sender, tx.SystemFee + tx.NetworkFee)
-	//}
-	//validators := g.NEO.getNextBlockValidators(ic)
-	//var netFee util.Fixed8
-	//for _, tx := range ic.block.Transactions {
-	//	netFee += tx.NetworkFee
-	//}
-	//g.mint(ic, <primary>, netFee)
+	for _, tx := range ic.Block.Transactions {
+		absAmount := big.NewInt(int64(tx.SystemFee.FloatValue() + tx.NetworkFee.FloatValue()))
+		g.burn(ic, tx.Sender, new(big.Int).Neg(absAmount))
```

Comment:
Why `Neg` here? `burn` does its own `Neg`, so now instead of burning tokens you're actually minting them.

Reply:
`burn` also calls `addTokens`, which throws an exception in case when amount is negative. Can we remove this check from `addTokens`?
```
if sign := amount.Sign(); sign == -1 {
		panic("negative amount")
	} 
```

Reply:
Well, this check is wrong then and `burn` doesn't work (was never tested), so we need to fix this issue.

Reply:
Removed this check from `addTokens` and `Neg` from `burn`, so now it should works as expected. Also moved this to a separate commit.